onboard/catkin_ws/src/cv/scripts/utils.py

```diff
@@ -53,3 +54,298 @@ def visualize_detections(self, frame, detections):
+
+    def convert_to_ros_compressed_msg(self, image, compressed_format='jpg'):
+        """
+        Convert any kind of image to ROS Compressed Image.
+        """
+        ros_cmp = None
+        if type(image) == np.ndarray:
+            ros_cmp = self.convert_cv2_to_ros_compressed_msg(
+                image, compressed_format=compressed_format)
+        elif image._type == 'sensor_msgs/Image':
+            ros_cmp = self.convert_ros_msg_to_ros_compressed_msg(
+                image, compressed_format=compressed_format)
+        elif image._type == 'sensor_msgs/CompressedImage':
+            ros_cmp = image
+        else:
+            raise TypeError("Cannot convert type: " + str(type(image)))
+        return ros_cmp
+
+    def convert_depth_to_ros_msg(self, image):
+        ros_msg = None
+        if type(image) == np.ndarray:
+            ros_msg = self.convert_cv2_to_ros_msg(image,
+                                                  image_encoding='mono16')
+        elif image._type == 'sensor_msgs/Image':
+            image.encoding = '16UC1'
+            ros_msg = image
+        elif image._type == 'sensor_msgs/CompressedImage':
+            ros_msg = self.convert_compressedDepth_to_image_msg(image)
+        else:
+            raise TypeError("Cannot convert type: " + str(type(image)))
+        return ros_msg
+
+    def convert_depth_to_ros_compressed_msg(self, image, encoding):
+        ros_cmp = None
+        if type(image) == np.ndarray:
+            ros_cmp = self.convert_cv2_to_ros_compressed_msg(image,
+                                                             compressed_format='png')
+            if encoding == 'mono16':
+                ros_cmp.format = '16UC1; compressedDepth'
+            elif encoding == 'mono8':
+                ros_cmp.format = '8UC1; compressedDepth'
+            else:
+                raise TypeError("Cannot compress image with encoding: " + encoding)
+
+        elif image._type == 'sensor_msgs/Image':
+            image.encoding = encoding
+            ros_cmp = self.convert_ros_msg_to_ros_compressed_msg(
+                image,
+                image_encoding=encoding,
+                compressed_format='png')
+
+            if encoding == 'mono16':
+                ros_cmp.format = '16UC1; compressedDepth'
+            elif encoding == 'mono8':
+                ros_cmp.format = '8UC1; compressedDepth'
+            else:
+                raise TypeError("Cannot compress image with encoding: " + encoding)
+
+        elif image._type == 'sensor_msgs/CompressedImage':
+            ros_cmp = image
+        else:
+            raise TypeError("Cannot convert type: " + str(type(image)))
+        return ros_cmp
+
+    def convert_depth_to_cv2(self, image):
+        cv2_img = None
+        if type(image) == np.ndarray:
+            cv2_img = image
+        elif image._type == 'sensor_msgs/Image':
+            image.encoding = 'mono16'
+
+            encoding = image.encoding.split(";")[0]
```

Comment:
You set image.encoding right before this, so this will always result in "mono16"

Reply:
Removed assignment to image.encoding.